internal/server/multitherading.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"bytes"
+
+	"github.com/dicedb/dice/internal/cmd"
+	"github.com/dicedb/dice/internal/comm"
+	"github.com/dicedb/dice/internal/eval"
+	"github.com/dicedb/dice/internal/ops"
+	"github.com/dicedb/dice/internal/shard"
+	"github.com/twmb/murmur3"
+)
+
+func getShard(key string, n uint32) uint32 {
+	// Hash the key using MurmurHash3
+	hash := murmur3.Sum32([]byte(key))
+	// Apply modulo operation
+	return hash % n
+}
+
+type ShardingMeta struct {
+	Cmd     string
+	Breakup func(mgr *shard.ShardManager, redisCmd *cmd.RedisCmd, c *comm.Client) []cmd.RedisCmd
+	Gather  func(responses ...eval.EvalResponse) []byte
+}
+
+var (
+	MultithreadingCmds = map[string]ShardingMeta{}
+
+	pingCmdMeta = ShardingMeta{
+		Cmd:     "PING",
+		Breakup: scatterPING,
+		Gather:  gatherPING,
+	}
+)
+
+func init() {
+	MultithreadingCmds["PING"] = pingCmdMeta
+}
+
+func (s *AsyncServer) cmdsBreakup(redisCmd *cmd.RedisCmd, c *comm.Client) []cmd.RedisCmd {
+	val, ok := MultithreadingCmds[redisCmd.Cmd]
+	if !ok {
+		return []cmd.RedisCmd{*redisCmd}
+	}
+
+	return val.Breakup(s.shardManager, redisCmd, c)
+}
+
+func (s *AsyncServer) scatter(cmds []cmd.RedisCmd, c *comm.Client) {
```

Comment:
I feel the below makes more sense:
1) Have a logic to route "Global" commands to any shard.
2) Route commands that use key based routing to specific shards -

```
func (s *AsyncServer) scatter(cmds []cmd.RedisCmd, c *comm.Client) {
	shardCount := s.shardManager.GetShardCount()

	for _, cmd := range cmds {
		var id uint32

		if isGlobalCommand(cmd.Name) {
			randomID, err := getRandomShardID(shardCount)
			if err != nil {
				id = 0
			} else {
				id = randomID
			}
		} else {
			key := cmd.Args[0]
			id = getShard(key, uint32(shardCount))
		}

		// Single combined statement
		s.shardManager.GetShard(shard.ShardID(id)).ReqChan <- &ops.StoreOp{
			Cmd:      &cmd,
			WorkerID: "server",
			ShardID:  int(id),
			Client:   c,
		}
	}
}

func isGlobalCommand(cmdName string) bool {
	globalCommands := map[string]bool{
		"PING": true,
		"INFO": true,
		"SET": false,
		// Add other global commands as needed
	}
	return globalCommands[strings.ToUpper(cmdName)]
}

func getRandomShardID(shardCount int) (uint32, error) {
	max := big.NewInt(int64(shardCount))
	n, err := rand.Int(rand.Reader, max)
	if err != nil {
		return 0, err
	}
	return uint32(n.Int64()), nil
}
```